src/node_config.rs

```diff
@@ -80,39 +116,65 @@ impl BootstrapNode {
 }
 
 /// Config parsed from command line arguments.
-#[derive(Clone, PartialEq, Eq, Debug)]
-pub struct CliConfig {
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize)]
+pub struct NodeConfig {
     /// UDP address to run DHT node
+    #[serde(rename = "udp-address")]
+    #[serde(default)]
     pub udp_addr: Option<SocketAddr>,
     /// TCP addresses to run TCP relay
+    #[serde(rename = "tcp-addresses")]
+    #[serde(default)]
     pub tcp_addrs: Vec<SocketAddr>,
     /// DHT SecretKey
+    #[serde(skip_deserializing)]
     pub sk: Option<SecretKey>,
     /// True if the SecretKey was passed as an argument instead of environment
     /// variable. Necessary to print a warning since the logger backend is not
     /// initialized when we parse arguments.
+    #[serde(skip_deserializing)]
     pub sk_passed_as_arg: bool,
     /// Path to the file where DHT keys are stored.
+    /// When run with config, this field is required.
+    #[serde(rename = "keys-file")]
+    pub keys_file_config: String,
```

Comment:
Can we just define our deserialization function which will use string deserialization and wrap the result to Some? I think it's better than introducing a new field and cloning to the old one.

Reply:
Let's do it in another PR